README.md

```diff
@@ -5,6 +5,59 @@ This repository contains the implementation of Taiga in Rust.
 
 ⚠️ It is a WIP project and cannot be used in production yet ⚠️
 
+# Introduction
+
+Taiga is a shielded state transition protocol that allows applications 
+built on top of it to enjoy the advantages of fully shielded multi-party 
+state transitions 
+(hiding the application type, the data associated with it, involved 
+parties, etc) without giving up the application complexity.
+
+![img.png](./book/src/images/intro_taiga_app.png)
+
+[**Validity predicates**](./book/src/validity-predicates.md) are a key 
+component of 
+Taiga - applications built on top of Taiga use them to express the 
+application rules, 
+and Taiga makes sure that the rules are being followed with the help of 
+the [Action circuit](./book/src/action.md).
+
+The state transition is considered to be valid if validity predicates of 
+all involved parties are satisfied, and the Action circuit check passed 
+successfully.
+
+
+#### ZKP privacy
+
+Taiga uses zero-knowledge proof system Halo2 to hide the sensitive 
+information about the state transitions (application types, parties 
+involved, etc). 
+Transactions of different applications are indistinguishable from one 
+another and all applications benefit from the shared shielded pool of 
+transactions.
+
+#### UTXO model
+
+Taiga is based on the UTXO model - the application states are stored in 
+immutable objects called **notes**. 
+To update the state, an application (or anyone with sufficient rights) 
+would spend the notes containing the old state and create notes containing 
```

Comment:
Would it make sense to explain what spend is? E.g. replace `would spend the notes` with `would "spend" (verifiable invalidate) the notes`?